test/qa/test_fileshare.py

```diff
@@ -222,13 +223,40 @@ def test_fileshare_transfer(background: bool, peer_name: meshnet.PeerName):
 
     assert peer_transfer_id is not None, "transfer was not received by peer"
 
-    ssh_client.exec_command(f"nordvpn fileshare accept --path /tmp {peer_transfer_id}")
+    transfers_local = sh.nordvpn.fileshare.list(_tty_out=False)
+    assert "request sent" in fileshare.find_transfer_by_id(transfers_local, local_transfer_id)
+
+    transfers_remote = ssh_client.exec_command("nordvpn fileshare list")
+    assert "waiting for download" in fileshare.find_transfer_by_id(transfers_remote, peer_transfer_id)
+
+    transfer_progress_local = fileshare.TransferProgressValidationThread(local_transfer_id, fileshare.TransferState.UPLOADING, None)
+    transfer_progress_local.start()
+
+    transfer_progress_remote = fileshare.TransferProgressValidationThread(local_transfer_id, fileshare.TransferState.DOWNLOADING, ssh_client)
+    transfer_progress_remote.start()
+
+    if path_flag:
+        peer_filepath = "/tmp/"
+        t_progress_interactive = ssh_client.exec_command(f"nordvpn fileshare accept {background_accept} --path {peer_filepath} {peer_transfer_id}")
+    else:
+        peer_filepath = "~/Downloads/"
+        t_progress_interactive = ssh_client.exec_command(f"nordvpn fileshare accept {background_accept} {peer_transfer_id}")
+
+    transfer_progress_local.join()
+    transfer_progress_remote.join()
+
+    assert transfer_progress_local.transfer_progress_valid
+    assert transfer_progress_remote.transfer_progress_valid
 
     time.sleep(1)
 
-    peer_filepath = f"/tmp/{wdir.filenames[0]}"
-    output = ssh_client.exec_command(f"cat {peer_filepath}")
-    assert message in output
+    if not background_send:
+        assert fileshare.validate_transfer_progress(command_handle.stdout.decode())
+
+    if not background_accept:
+        assert fileshare.validate_transfer_progress(t_progress_interactive)
+
+    ssh_client.exec_command(f"ls {peer_filepath}/{wdir.filenames[0]}") # this will fail, if file is not found
```

Comment:
I think it would make sense to check for file contents as well to make sure transfer was actually successful